SalesforceCommerceCloud/pwa-kit
# Hide the wishlist icon when there are not event handlers for it

**Problem**: There is a bug reported that the wishlist icon is not working in the Recommended Items and Recently Viewed section. It is because the event handlers for wishlist is in the component `ProductTile` component that is being used by Recommended Items and Recently Viewed is not defined.

**Solution**: We should guard the rendering of wishlist icon. It should be hidden if `onAddToWishlistClick` and `onRemoveWishlistClick` is not defined. 

 **JIRA**: (link to JIRA ticket)
 **Linked PRs**: (links to corresponding PRs, optional)

## How to test-drive this PR
- `npm start`
- Go to PDP page
- Scroll down to the bottom
- You should not see the wishlist icon on the products in Recommended Items and Recently view section

## Changes
- `product-tile`

## Todos

### General

### Backwards Compatibility

> Breaking changes include:
> * Removing a public function/Component/Prop
> * Adding required arguments to functions
> * Changing the data-types of function params or return values

### Accessibility Compliance
You must check off all items in **one** of the follow two lists:

_or..._


### Documentation


> Documentation and comments should follow the [Mobify Content Style Guide](https://docs.google.com/a/mobify.me/document/d/1jlcg5boC3MUHN7fy2n3Yu_FBGzGgTUPrnMoePzQtqE4/edit?usp=sharing) and [Mobify's Capitalization and Spelling List](https://docs.google.com/document/d/1LO7RAr2vD3LFs_bj5j0vIFMKsVFrOFfEjpCheYEH3Lg/edit?usp=sharing)
